src/main/java/com/playonlinux/domain/lang/LanguageBundleSelector.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2015 Jonas Konrad
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package com.playonlinux.domain.lang;
+
+import com.playonlinux.domain.Localisation;
+import java.io.IOError;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Locale;
+import org.fedorahosted.tennera.jgettext.PoParser;
+
+public class LanguageBundleSelector {
+    /**
+     * Select a LanguageBundle for the given resource from the classpath.
+     *
+     * @return The language bundle, never null. May be a {@link FallbackLanguageBundle} which will do a no-operation
+     * translation.
+     */
+    public static LanguageBundle forLocale(Locale locale) {
+        // first, try the full language-country tag (either xx-YY or just xx if no country was specified)
+        String fullTag = new Locale(locale.getLanguage(), locale.getCountry())
+                .toLanguageTag().replace('-', '_');
+        LanguageBundle bundle = forLocaleIdOrNull(fullTag);
+        if (bundle != null) { return bundle; }
+
+        // then, try the base language tag (just xx for language)
+        String languageTag = new Locale(locale.getLanguage())
+                .toLanguageTag().replace('-', '_');
+        bundle = forLocaleIdOrNull(languageTag);
+        if (bundle != null) { return bundle; }
+
+        // if neither was found, fall back on defaults
+        return FallbackLanguageBundle.getInstance();
+    }
+
+    private static CatalogLanguageBundle forLocaleIdOrNull(String localeId) {
+        URL poResourceUrl = Localisation.class.getResource("/locale/po/" + localeId + ".po");
+        if (poResourceUrl == null) {
+            return null;
+        } else {
+            return parseCatalogBundle(poResourceUrl);
+        }
+    }
+
+    private static CatalogLanguageBundle parseCatalogBundle(URL poResourceUrl) {
+        PoParser parser = new PoParser();
+        try (InputStream inputStream = poResourceUrl.openStream()) {
+            return CatalogLanguageBundle.fromCatalog(parser.parseCatalog(inputStream, false));
+        } catch (IOException e) {
+            // this should only happen if a PO file is broken so we throw an Error
```

Comment:
Actually, it should only happen when theres an FS / zip / other classloader issue:

- Missing PO is checked above in `poResourceUrl == null`
- Broken PO will throw a `org.fedorahosted.tennera.jgettext.catalog.parse.ParseException` which will escalate anyway